### PR TITLE
change topic number to topicID

### PR DESCRIPTION
**Pull request by povolev15**
change topic number to topicID in protobuf

https://github.com/hashgraph/hedera-services/issues/7104

### CODE REVIEW

**Review by kimbor — COMMENTED** [review]
It should be:
TopicID topic_id
in order to follow the naming convention that other places in hedera-protobufs follows.
Also, you should update the javadoc to say id instead of number.

**Comment by jsync-swirlds** [comment]
There are exactly two ID values in state prior to this PR, both are `id`.
The ID values in the public transaction protobufs often refer to an object other than the current, or are expanded to be clearer when translated to other languages (e.g. typescript where `id` can be a problem).
We should probably discuss in the daily team call which way (`name.name_id` or `name.id`) we want to go in the states objects.

File: services/state/consensus/topic.proto [hunk 1]
```diff
@@ -53,7 +53,7 @@ message Topic {
     /**
      * The topic's unique entity number in the Merkle state.
```
Comment (jsync-swirlds) [inline services/state/consensus/topic.proto:54]: `s/number/id/`
Note, this PR cannot merge before the services PR that mirrors it.

**Review by iwsimon — APPROVED** [review]
LGTM

File: services/state/consensus/topic.proto (hunk 1 above)
Comment (povolev15) [reply services/state/consensus/topic.proto:54]: this one https://github.com/hashgraph/hedera-services/pull/7216

**Comment by povolev15** [comment]
restored the branch that it will not break develop

**Pull request merged**